### ISSUE TITLE
[Issue] Crawl github repositories

### ISSUE THREAD

**Issue by dasheck0** [StanGirard/quivr#942]
## Submitting a Feature Request 💡

Feature requests are welcome. But take a moment to find out whether your idea fits with the scope and aims of the project. It's up to you to make a strong case to convince the project's developers of the merits of this feature. Please provide as much detail and context as possible.

**Example:**

Support crawling public repositories in github. There are many awesome repositories (I mean literally awesome i.e. https://github.com/mehdihadeli/awesome-software-architecture). Would be great to provide a url and quivr can recursively crawl the contents. Also many open source projects provide documentation on github, which would be another great use case. And finally crawling any kind of repository with some code comprehension would be great as you could talk to your repo as well. Great for developers.

IMPLEMENTATION. If possible, provide some details on how you think this could be implemented.

Basically the url crawler is a great start. It should work properly (as of now there are some bugs). There are some libraries out there in the javascript eco system like (cheerio) that help dealing with the DOM. This way it would be easy to get all the a tags that refer to other pages. One have to deal with branching here, since one would crawl too much at some point (maybe checking for the same domain or providing a fetch depth?). Just some ideas. With this approach probably related to #926
